packages/s-deploy/src/deploy.ts

```diff
@@ -0,0 +1,59 @@
+import { lodash as _, Logger, fs } from '@serverless-cd/core';
+import execa from 'execa';
+
+export interface IProps {
+  resource?: string; // Specify a resource in s.yaml. Use the command to s xxx deploy.
+  deployFile?: string; // Specific yaml file. Use the command to s deploy -t xxx 
+  deployArgs?: string[]; // Specific custom args. Merge with other command args during deployment.
+  envName?: string; // Specify envionment. Use the command to s deploy --env
+  debugMode?: boolean; // Use the command to s deploy --debug
+  worksapce?: string // worksapce dir.
+}
+
+class Deploy {
+  private logger: Logger;
+  private props: IProps;
+  constructor(props: IProps, logger: Logger) {
+    this.logger = (logger || console) as Logger;
+    this.props = props
+  }
+
+  async run() {
+    const { resource, deployFile = 's.yaml', deployArgs, envName, debugMode, worksapce } = this.props;
+    if (!fs.existsSync(`${worksapce}/${deployFile}`)) {
```

Comment:
what if `deployFile` is a relative path. e.g. './resources/s.yaml'.
maybe we should use `path.resolve(workspace, deployFile)` instead.